discord/ui/container.py

```diff
@@ -0,0 +1,371 @@
+
+    This can be inherited.
+
+
+    .. versionadded:: 2.6
+
+    Examples
+    --------
+
+    .. code-block:: python3
+
+        import discord
+        from discord import ui
+
+        # you can subclass it and add components as you would add them
+        # in a LayoutView
+        class MyContainer(ui.Container):
+            action_row = ui.ActionRow()
+
+            @action_row.button(label='A button in a container!')
+            async def a_button(self, interaction: discord.Interaction, button: discord.ui.Button):
+                await interaction.response.send_message('You clicked a button!')
+
+        # or use it directly on LayoutView
+        class MyView(ui.LayoutView):
+            container = ui.Container(ui.TextDisplay('I am a text display on a container!'))
+            # or you can use your subclass:
+            # container = MyContainer()
+
+    Parameters
+    ----------
+    \*children: :class:`Item`
+        The initial children of this container.
+    accent_colour: Optional[Union[:class:`.Colour`, :class:`int`]]
+        The colour of the container. Defaults to ``None``.
+    accent_color: Optional[Union[:class:`.Colour`, :class:`int`]]
+        The color of the container. Defaults to ``None``.
+    spoiler: :class:`bool`
+        Whether to flag this container as a spoiler. Defaults
+        to ``False``.
+    id: Optional[:class:`int`]
+        The ID of this component. This must be unique across the view.
+    """
+
+    __container_children_items__: ClassVar[Dict[str, Union[ItemCallbackType[Any], Item[Any]]]] = {}
+    __discord_ui_container__: ClassVar[bool] = True
+    __item_repr_attributes__ = (
+        'accent_colour',
+        'spoiler',
+        'id',
+    )
+
+    def __init__(
+        self,
+        *children: Item[V],
+        accent_colour: Optional[Union[Colour, int]] = None,
+        accent_color: Optional[Union[Color, int]] = None,
+        spoiler: bool = False,
+        id: Optional[int] = None,
+    ) -> None:
+        super().__init__()
+        self._children: List[Item[V]] = self._init_children()
+        for child in children:
+            self.add_item(child)
+
+        self.spoiler: bool = spoiler
+        self._colour = accent_colour if accent_colour is not None else accent_color
+        self.id = id
+
+    def __repr__(self) -> str:
+        return f'<{self.__class__.__name__} children={len(self._children)}>'
+
+    def _init_children(self) -> List[Item[Any]]:
+        children = []
+        parents = {}
+
+        for name, raw in self.__container_children_items__.items():
+            if isinstance(raw, Item):
+                item = copy.deepcopy(raw)
+                item._parent = self
+                if getattr(item, '__discord_ui_section__', False) and item.accessory.is_dispatchable():  # type: ignore
+                    if item.accessory._provided_custom_id is False:  # type: ignore
+                        item.accessory.custom_id = os.urandom(16).hex()  # type: ignore
```

Comment:
I do not understand why this is here still. The custom ID stuff is the responsibility of the class itself, not the outside.